presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -255,15 +255,28 @@ public ConnectorSplitSource getSplits(ConnectorTransactionHandle transaction, Co
                     }
                 }
 
-                Optional<HiveBucketProperty> partitionBucketProperty = partition.getStorage().getBucketProperty();
-                checkCondition(
-                        partitionBucketProperty.equals(bucketProperty),
-                        HiveErrorCode.HIVE_PARTITION_SCHEMA_MISMATCH,
-                        "Hive table (%s) bucketing property (%s) does not match partition (%s) bucketing property (%s)",
-                        hivePartition.getTableName(),
-                        bucketProperty,
-                        hivePartition.getPartitionId(),
-                        partitionBucketProperty);
+                if (bucketProperty.isPresent()) {
```

Comment:
Can you extract everything within this `if` block to separate function, e.g. `ensureBucketingMatch`, to make this method more succinct?